### PR TITLE
Add and improve docs for sugar3.graphics.objectchooser

### PR DESCRIPTION
**Pull request by mjuvekar7**
Add docs and improve previously present docstrings in sugar3.graphics.objectchooser module.

### CODE REVIEW

**Comment by mjuvekar7** [comment]
for GCI task at https://codein.withgoogle.com/dashboard/task-instances/5466829778583552/

File: src/sugar3/graphics/objectchooser.py [hunk 1]
```diff
@@ -42,16 +42,19 @@
...
+    """
+    Retrive a pixbuf with the content of the preview field
 
-    Keyword arguments:
-    metadata -- the metadata dictionary.
-                Can't be None, use metadata.get('preview', '')
-    width -- the pixbuf width, if is not set, the default width will be used
-    height -- the pixbuf width, if is not set, the default height will be used
+    Args:
+        metadata (dictionary): the metadata dictionary. Can't be None, use metadata.get('preview', '')
```
Comment (samdroid-apps) [inline src/sugar3/graphics/objectchooser.py]: Maybe put a usage example that uses the `metadata.get('preview;, '')` after the summary (2 lines above)

File: src/sugar3/graphics/objectchooser.py [hunk 2]
```diff
@@ -99,22 +102,20 @@ def get_preview_pixbuf(preview_data, width=-1, height=-1):
 
 
 class ObjectChooser(object):
+    """
+    UI interface for objec choosers.
```
Comment (samdroid-apps) [inline src/sugar3/graphics/objectchooser.py]: typo "objec"

File: src/sugar3/graphics/objectchooser.py [hunk 3]
```diff
@@ -99,22 +102,20 @@ def get_preview_pixbuf(preview_data, width=-1, height=-1):
...
-                 show_preview=False):
-        """Initialise the ObjectChoser
-
-        parent -- the widget calling ObjectChooser
+    Object choosers can be used by acivities to allow the user to select objects
+    from the file system or from some other similar source.
 
-        what_filter -- string
+    Keyword Args:
+        parent (:class:`Gtk.Widget`): the widget calling ObjectChooser
```
Comment (samdroid-apps) [inline src/sugar3/graphics/objectchooser.py]: What should this be usually?  Should it be the sugar3.activity.activity.Activity (self for some activities)?  Maybe give a usage example?

File: src/sugar3/graphics/objectchooser.py [hunk 4]
```diff
@@ -99,22 +102,20 @@ def get_preview_pixbuf(preview_data, width=-1, height=-1):
...
 
-            string should be an activity bundle_id or a generic mime
-            type as defined in mime.py used to determine which objects
+        what_filter (str): an activity bundle_id or a generic mime type as
+            defined in mime.py used to determine which objects
```
Comment (samdroid-apps) [inline src/sugar3/graphics/objectchooser.py]: Please replace mime.py with the appropriate `:class:` (or whatever for modules) so Sphinx makes a link.

File: src/sugar3/graphics/objectchooser.py [hunk 5]
```diff
@@ -170,6 +171,13 @@ def __init__(self, parent=None, what_filter=None, filter_type=None,
         self._filter_type = filter_type
 
     def run(self):
+        """
+        Runs the object chooser and displays it.
+        
+        Returns:
+            Gtk.ResponseType constant, the response received from displaying the
+                object chooser.
```
Comment (samdroid-apps) [inline src/sugar3/graphics/objectchooser.py]: What response types does it return?  What do they mean in the context of the chooser?

**Comment by samdroid-apps** [comment]
Looks good.

Would be great to put maybe an example of using an object chooser at the top of the file?  You should find an example in an activity, please use the embed method like in line 26 of https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/269/files

**Comment by godiard** [comment]
You can get examples from here http://godiard.blogspot.com.ar/2013/07/sugar-programming-improovements-in.html if you want

**Comment by mjuvekar7** [comment]
Could not find any usage example of the object chooser (tried using grep too)

**Comment by mjuvekar7** [comment]
Thanks @godiard that was really helpful!

**Comment by quozl** [comment]
Reviewed.

**Comment by samdroid-apps** [comment]
Great, will merge.

**Pull request merged**